packages/cmf/src/sagaRouter/documentTitle.js

```diff
@@ -0,0 +1,82 @@
+import { take } from 'redux-saga/effects';
+
+import matchPath from './matchPath';
+
+/**
+ * Format the current full path
+ * @param {string} path current path name (ex: foo)
+ * @param {string} parentPath parent path name (ex: /foo)
+ */
+export function formatPath(path, parentPath) {
+	let fPath = path;
+	if (parentPath) {
+		fPath = parentPath.endsWith('/') ? `${parentPath}${path}` : `${parentPath}/${path}`;
+	}
+	// Convert optional url parameters to React Router V2 --> V4 (:param) --> :param?
+	return fPath.replace(/[(]/g, '').replace(/[)]/g, '?');
+}
+
+/**
+ * Build a map from the settings routes object recursively
+ * @param {object} routes
+ * @param {array} mapRoutes
+ * @param {string} parentPath
+ */
+export function buildMapFromRoutes(routes, mapRoutes, parentPath) {
+	const path = formatPath(routes.path, parentPath);
+	if (routes.documentTitle) {
+		mapRoutes.set(path, routes.documentTitle);
+	}
+	const childRoutes = routes.childRoutes;
+	if (childRoutes && childRoutes instanceof Array) {
+		childRoutes.forEach((route) => {
+			buildMapFromRoutes(route, mapRoutes, path);
+		});
+	}
+	return mapRoutes;
+}
+
+/**
+ * Iterate over the routes map to find a match with the given location.
+ * The value map return the correct document title.
+ * If none is found we return the default document title.
+ * @param {array} mapRoutes
+ * @param {string} location
+ * @param {string} defaultDocTitle
+ */
+export function getTitleFromRoutes(mapRoutes, location, defaultDocTitle) {
+	let title = defaultDocTitle;
+	mapRoutes.forEach((value, key) => {
```

Comment:
I prefer use find instead forEach. As soon as it finds the value, the loop stops instead to look on all values.

Also, to the ret const, use a name more comprehensible.